### ISSUE TITLE
impossible issue

### ISSUE THREAD

**Issue by DjSapsan** [AoE2CommunityGitHub/WololoKingdoms#32]
Im tried to install WK from very beginning. It always was superbugged on all versions. Actually i had some problems with voobly too. Problem is that i have different paths. Lets me explain.
1) I have 3 physical disks and 5 logical. Long time ago my AoE2HD and voobly was installed on first.
2) After i reinstalled system few times and got 2 more disks. In that time installation of AoE2HD and voobly moved from one folder to folder many times (i mean clean installation)
3)

**Comment by DjSapsan** [comment]
P.S. I fixed minor voobly issue. I can run game on voobly. But WK still cant find game

**Comment by DjSapsan** [comment]
any answer?

**Comment by gregstein** [comment]
Compatibility Patch is the issue. Try to setup Voobly like this https://www.youtube.com/watch?v=Srhi14fjTkM

**Comment by Jineapple** [comment]
The WK installer checks the registry for the steam installation directory. If it can't find the registry keys, it checks the folder the installer is placed in (so you should try placing it in the Age2HD directory). If it still doesn't find a game installation, it fails with the error you got.

If placing the converter in the Age2HD directory doesn't work, can you check a couple of things?

Can you check if HKEY_LOCAL_MACHINE\Software\WOW6432Node\Valve\Steam (64-bit)
or HKEY_LOCAL_MACHINE\Software\Valve\Steam (32-bit) exists in your registry and points to the right directory?

From the directory you've given, it sounds like your steam library is installed somewhere else than steam itself. The installer should be able to read this from steam files and identify the steam library directory, but it's possible that there's a bug here.

If this is the case, can you check that you have a <steam installation path>/steamapps/libraryfolders.vdf file and that this file contains the "X:\SteamLibrary\steamapps\" path and paste the contents here?

**Comment by DjSapsan** [comment]
I tried to fix it by myself many times, but in the end I just gave up. I will never use WK and voobly anymore. I cant waste time anymore.
От: Jineapple
Отправлено: 28 мая 2019 г. в 16:23
Кому: AoE2CommunityGitHub/WololoKingdoms
Копия: Sapsan; Author
Тема: Re: [AoE2CommunityGitHub/WololoKingdoms] impossible issue (#32)

The WK installer checks the registry for the steam installation directory. If it can't find the registry keys, it checks the folder the installer is placed in (so you should try placing it in the Age2HD directory). If it still doesn't find a game installation, it fails with the error you got.
If placing the converter in the Age2HD directory doesn't work, can you check a couple of things?
Can you check if HKEY_LOCAL_MACHINE\Software\WOW6432Node\Valve\Steam (64-bit)
or HKEY_LOCAL_MACHINE\Software\Valve\Steam (32-bit) exists in your registry and points to the right directory?
From the directory you've given, it sounds like your steam library is installed somewhere else than steam itself. The installer should be able to read this from steam files and identify the steam library directory, but it's possible that there's a bug here.
If this is the case, can you check that you have a /steamapps/libraryfolders.vdf file and that this file contains the "X:\SteamLibrary\steamapps" path and paste the contents here?

**Comment by Jineapple** [comment]
I'm sorry to hear that. I've tried to make the installer as robust as possible, but it's hard to predict/test different installation configurations that people might have. Stuff like this mostly can only be fixed once there's reports of it that help isolate the problem.
I can understand the frustration of course. But if you find the time in the future to to try again and provide feedback on what does and doesn't work, maybe this can be fixed.

**Comment by gregstein** [comment]
@Jineapple 
Why don't you grab the installation path of steam through Steamworks just the way I have implemented through AoE2Tools? 

I do not rely on registry to grab the game path because obviously the majority of users install their games on a different location.

In my WK Cloud Installer & Ao2Tools Cloud I grab the installation path natively using **GetAppInstallDir** I am not a c++ developer but here is a clue from my c# code:


https://github.com/gregstein/AoE2Tools/blob/9ae1a2c67659dd2a9ce622c6739a842bf7826978/Game%20Tweaker/bin/Debug/Steamworks.NET-StandaloneTest-master/Steamworks.NET-StandaloneTest/WK%20Cloud%20Installer.cs#L167-L184

**Comment by DjSapsan** [comment]
@gregstein actually, after i describe my problem i have used your AoE2Tools to install WK properly and i could launch singleplayer game. But somehow it broke the voobly and i could not join/create any room because of different error. After i got this error i gave up because it doesnt worth efforts.

**Comment by gregstein** [comment]
Hello DjSapsan 

Let me clarify a little something. I have had many users reporting issues related to AoE2Tools /Voobly/Game installation and not a single issue slipped away from me without providing a solid solution to it. 

I noticed that you have been talking about errors but you have failed to be more specific. 

Is it "Directdraw error", is it "Could not join" error, etc.? 

AoE2Tools has never broken or changed the way voobly operates. AoE2Tools simply downloads Voobly client, silent install it and change gameroom settings. Technically that's not breaking functionality. You should probably be more specific how it broke Voobly so i can assist you further.

Also the manner you are addressing Jineapple and I, must be improved otherwise in the future we would just ignore your queries. 

If you are struggling with something you could contact either one of us through Discord/Voobly/Reddit and we would resolve those issues for you.

Warm regards,

**Comment by DjSapsan** [comment]
Yes, I got directdraw issue. After I fixed it I got another error I don’t remember exactly.

Отправлено из приложения "Почта" для Windows 10

От: gregstein
Отправлено: 29 травня 2019 р. в 19:48
Кому: AoE2CommunityGitHub/WololoKingdoms
Копия: Sapsan; Comment
Тема: Re: [AoE2CommunityGitHub/WololoKingdoms] impossible issue (#32)

Hello DjSapsan
Let me clarify a little something. I have had many users reporting issues related to AoE2Tools /Voobly/Game installation and not a single issue slipped away from me without providing a solid solution to it.
I noticed that you have been talking about errors but you have failed to be more specific.
Is it "Directdraw error", is it "Could not join" error, etc.?
AoE2Tools has never broken or changed the way voobly operates. AoE2Tools simply downloads Voobly client, silent install it and change gameroom settings. Technically that's not breaking functionality. You should probably be more specific how it broke Voobly so i can assist you further.
Also the manner you are addressing Jineapple and I, must be improved otherwise in the future we would just ignore your queries.
If you are struggling with something you could contact either one of us through Discord/Voobly/Reddit and we would resolve those issues for you.
Warm regards,

**Comment by DjSapsan** [comment]
I fixed directdraw with “healing” option, but got another error. After another error I just deleted everything and don’t want to get back to this. Bye

Отправлено из приложения "Почта" для Windows 10

От: gregstein
Отправлено: 29 травня 2019 р. в 20:55
Кому: AoE2CommunityGitHub/WololoKingdoms
Копия: Sapsan; Comment
Тема: Re: [AoE2CommunityGitHub/WololoKingdoms] impossible issue (#32)

The fix for Directdraw error is integrated within AoE2Tools see resolved ticket gregstein/AoE2Tools#18
Go on feed me more issues and I will help you fix them one by one just provide me with the error exception.

**Comment by gregstein** [comment]
You gave up way too early brother 11
